xdebug_stack.c

```diff
@@ -1288,6 +1291,22 @@ function_stack_entry *xdebug_add_stack_frame(zend_execute_data *zdata, zend_op_a
 					} else {
 						ZVAL_COPY_VALUE(&(tmp->var[tmp->varc].data), ZEND_CALL_VAR_NUM(zdata, zdata->func->op_array.last_var + zdata->func->op_array.T + i - arguments_wanted));
 					}
+					if (Z_TYPE(tmp->var[tmp->varc].data) == IS_OBJECT) {
+						/* Tell garbage collector to NOT gc the objects until the stack is removed
+						 * Otherwise, Zend GC might remove functions parameters (or maybe other variables content)
+						 * and XDebug will be stuck when trying to dump the stack frame
+						 * because some variables in this stackframe will no longer exists (because being GCed) */
+						#if PHP_VERSION_ID >= 70300
```

Comment:
The `#if …` lines should start in column 1 of the line, with the `GC_ADDREF`/`GC_REFCOUNT` lines not extra indented.

I saw somewhere that you had problems with `GC_ADDREF` and PHP 7.3 ? Is that now solved?

Reply:
Indent fixed (same at thje 2nd place, L1410)

Indeed: GC_ADDREF is PHP7.3+ so Github tests failed on PHP 7.1/7.2, that's why I've used the #if directive to use GC_REFCOUNT instead. GC_ADDREF was added by this commit https://github.com/php/php-src/commit/49ea143bbd8d0bfcd393d0b5308a5d7833fc087c in PHP7.3
I might remove the #if things but then, xdebug 2.8 (aka once PR is merged) will be only PHP 7.3+ compatible?

Reply:
Xdebug 2.7 and 2.8 should all still work with PHP 7.0 to PHP 7.3 (for 2.7) and PHP 7.4 (for 2.8). I've no plans to drop support for PHP 7.0 until Xdebug 3.

Reply:
Alright, I'll keep the precompiler test then